src/chromium/interfaces/launchBrowser.js

```diff
@@ -1,33 +1,57 @@
 import { isURLValid } from "Shared/backgroundScripts/validTab.js";
-import { getIsFirefoxInstalled } from "./getters.js";
+import { getInstalledFirefoxVariant } from "./getters.js";
+import { failedBrowserLaunch } from "Shared/generated/launchEvent.js";
 
 /**
  * Launches the Firefox variant. If Firefox is not installed, launch the Firefox download page.
  *
  * @param {string} url The url to launch in the browser.
- * @param {boolean} usePrivateBrowsing  True if firefox should be launched in private browsing mode, false for normal mode.
- * @returns {boolean} True if the browser was launched, false otherwise.
+ * @param {boolean} usePrivateBrowsing True if firefox should be launched in private browsing mode, false for normal mode.
+ * @returns {Promise<boolean>} A promise that resolves to true if the browser was launched, false otherwise.
  */
 export async function launchBrowser(url, usePrivateBrowsing = false) {
   if (!isURLValid(url)) {
     console.error("Invalid URL scheme:", url);
     return false;
   }
 
-  if (!(await getIsFirefoxInstalled())) {
-    browser.tabs.create({ url: "https://www.mozilla.org/firefox/" });
+  const nativeMessagingHost = await getInstalledFirefoxVariant();
+  if (!nativeMessagingHost) {
+    // No Firefox variant is installed. Direct the user to the Firefox download page.
+    await browser.tabs.create({
+      url: "https://www.mozilla.org/firefox/new/",
```

Comment:
Did you mean to use welcomePageNoExternalBrowserErrorChromium here?
I can't see what that localized value is used and it does match here.

Reply:
That value is used solely in the welcome page for the text inside the error notification. Right here, it seems that I missed the change in the [spec 1.5.4](https://docs.google.com/document/d/1jTxG4kyDsgwAVHYkDvHvABmO85-wk9vppJe6ikNe76A/edit#heading=h.dfbjsayo38mn) that if the user attempts to launch the browser with none installed it redirects to the welcome page. It used to say to launch the firefox install page.